shared/lib-hoppy/hoppy/async_consumer.py

```diff
@@ -0,0 +1,195 @@
+
+    def open_channel(self):
+        logging.debug('Consumer - Creating a new channel')
+        self._connection.channel(on_open_callback=self.on_channel_open)
+
+    def on_channel_open(self, channel):
+        logging.debug('Consumer - Channel opened')
+        self._channel = channel
+        self._channel.add_on_close_callback(self.on_channel_closed)
+        self.setup_exchange(self.exchange)
+
+    def on_channel_closed(self, channel, reason):
+        logging.warning(f'Consumer - Channel {channel} was closed: {reason}')
+        self.close_connection()
+
+    def setup_exchange(self, exchange_name):
+        logging.debug(f'Consumer - Declaring exchange: {self.exchange}')
+        self._channel.exchange_declare(
+            exchange=exchange_name,
+            exchange_type=self.exchange_type,
+            durable=True,
+            auto_delete=True,
+            callback=self.on_exchange_declare_ok)
+
+    def on_exchange_declare_ok(self, _unused_frame):
+        logging.debug(f'Consumer - Exchange declared: {self.exchange}')
+        self.setup_queue()
+
+    def setup_queue(self):
+        logging.debug(f'Consumer - Declaring queue {self.queue}')
+        self._channel.queue_declare(queue=self.queue, callback=self.on_queue_declare_ok)
+
+    def on_queue_declare_ok(self, _unused_frame):
+        logging.debug(f'Consumer - Binding {self.exchange} to {self.queue} with {self.routing_key}')
+        self._channel.queue_bind(
+            self.queue,
+            self.exchange,
+            routing_key=self.routing_key,
+            callback=self.on_bind_ok)
+
+    def on_bind_ok(self, _unused_frame):
+        logging.debug(f'Consumer - Queue bound: {self.queue}')
+        self._channel.basic_qos(prefetch_count=self._prefetch_count, callback=self.on_basic_qos_ok)
+
+    def on_basic_qos_ok(self, _unused_frame):
+        logging.debug(f'Consumer - QOS set to: {self._prefetch_count}')
+        self.start_consuming()
+
+    def start_consuming(self):
+        logging.debug('Consumer - Issuing consumer related RPC commands')
+        self._channel.add_on_cancel_callback(self.on_consumer_cancelled)
+        self._consumer_tag = self._channel.basic_consume(self.queue, self.on_message)
+        self._consuming = True
+
+    def on_consumer_cancelled(self, method_frame):
+        logging.debug(f'Consumer - Consumer was cancelled remotely, shutting down: {method_frame}')
+        if self._channel:
+            self._channel.close()
+
+    def on_message(self, _unused_channel, basic_deliver, properties, body):
+        logging.debug(f'Consumer - Received message # {basic_deliver.delivery_tag} from {properties.app_id}: {body}')
+        self.reply_callback(self._channel,
+                            properties,
+                            basic_deliver.delivery_tag,
+                            body)
+
+    def acknowledge_message(self, delivery_tag):
+        logging.debug('Consumer - Acknowledging message %s', delivery_tag)
```

Comment:
what is the reason for using .debug, should we use .info instead? Just curious

Reply:
I set it to default because these classes can generate tons of logs, so i figure we can change the log level if there's an issue that is noticed.